scripts/OPNFV/NetworkServices/Methods.class/__methods__/deploy_cfns.rb

```diff
@@ -89,27 +90,65 @@ def base_template(name)
   }
 end
 
-def create_template(name)
-  vpc_id = 'vpc-a06de3c5' 
-  security_group_1 = "ControlSecurityGroup"
-  security_group_2 = "WebSecurityGroup"
-  nic1 = "ControlXface"
-  nic2 = "WebXface"
-  
+def create_template(name,nsd_properties,nsd_requirements)
+  vpc_id = nsd_properties['vpc_id'] #'vpc-303bc157' 
+  image_id = nsd_properties['image_id']
+  subnet=nil
+  security_group=nil
+  availability_zone=nsd_properties['availabilityzone']
+  key_name=nsd_properties['key_name']
+  instance_type=nsd_properties['image_type']
   template_content = base_template(name)
-  template_content["Resources"].merge!(security_group(security_group_1, vpc_id))
-  template_content["Resources"].merge!(security_group(security_group_2, vpc_id))
-  template_content["Resources"].merge!(template_interface(nic1, 'subnet-ac904787', security_group_1))
-  template_content["Resources"].merge!(template_interface(nic2, 'subnet-1852bb33', security_group_2)) 
-  template_content["Resources"].merge!(instance("Ec2Instance", [nic1, nic2], "EmsRefreshSpec-KeyPair", 'ami-5769193e'))
+  nic = Array.new 
+  
+  network=$evm.vmdb('ManageIQ_Providers_Amazon_NetworkManager_CloudNetwork').find_by_name(vpc_id.strip)
+  image=$evm.vmdb('ManageIQ_Providers_Amazon_CloudManager_Template').find_by_name(image_id.strip)
+  if network==nil || image==nil
+    $evm.log(:error,"vpc_id #{vpc_id} or image id #{image_id} for amazon is not valid.")
+     exit MIQ_ERROR
```

Comment:
we would also need to set $evm.root['ae_result'] = 'error' here

Best is to just call 

raise "vpc_id #{vpc_id} or image id #{image_id} for amazon is not valid."

then the logic for exiting state machine is at the bottom